spec/controllers/voice_controller_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe VoiceController do
+  describe '#otp' do
+    subject(:action) { get :otp, code: code, repeat_count: repeat_count, format: :xml }
+    let(:code) { nil }
+    let(:repeat_count) { nil }
+
+    context 'without a code in the URL' do
+      let(:code) { nil }
+
+      it 'renders a blank 400' do
+        action
+
+        expect(response).to be_bad_request
+        expect(response.body).to be_empty
+      end
+    end
+
+    context 'with a code in the URL' do
+      render_views
+
+      let(:code) { 1234 }
+
+      it 'tells Twilio to <Say> the code with pauses in between' do
+        action
+
+        doc = Nokogiri::XML(response.body)
+        say = doc.css('Say').first
+        expect(say.text).to include('1, 2, 3, 4,')
+      end
+
+      it 'has a <Gather> with instructions to repeat with a repeat_count' do
+        action
+
+        doc = Nokogiri::XML(response.body)
+        gather = doc.css('Gather').first
+
+        expect(gather['action']).to include('repeat_count=4')
+      end
+
+      context 'when repeat_count is too low' do
```

Comment:
So we want to limit the # of times the call can be looped, so I track this state via a URL parameter. When the the `repeat_count` tracks the value and counts down, so maybe I just should have specified "when the repeat_count gets to 1" -- will fix